presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSource.java

```diff
@@ -57,11 +57,15 @@
 
     private final FileFormatDataSourceStats stats;
 
+    // Row ID relative to all the original files of the same bucket ID before this file in lexicographic order
+    private Optional<Long> originalFileRowId = Optional.empty();
```

Comment:
Thanks for commenting here. 

However, i do not fully understand. The original files do not have row ids, do they?

Reply:
Yes, original files don't have row ids, but we need to calculate it to filter out the deleted rows as mentioned at line: L121.